src/tests/EShopworld.WorkerProcess.IntegrationTests/DistributedLockTests.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Eshopworld.Core;
+using Eshopworld.DevOps;
+using Eshopworld.Telemetry;
+using Eshopworld.Tests.Core;
+using EShopworld.WorkerProcess.Configuration;
+using EShopworld.WorkerProcess.CosmosDistributedLock;
+using FluentAssertions;
+using Microsoft.ApplicationInsights;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace EShopworld.WorkerProcess.IntegrationTests
+{
+    public class DistributedLockTests
+    {
+        private readonly ServiceProvider _serviceProvider;
+        private readonly ITestOutputHelper _output;
+
+        public DistributedLockTests(ITestOutputHelper output)
+        {
+            var configuration = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile("appsettings.json", false, true)
+                .Build();
+
+            var serviceCollection = new ServiceCollection();
+            ConfigureServices(serviceCollection, configuration);
+
+            _serviceProvider = serviceCollection.BuildServiceProvider();
+
+            _output = output;
+        }
+
+        [Fact, IsIntegration]
+        public void AcquireLock_ForNewLock_ShouldNotThrowException()
+        {
+            var disLockStore = _serviceProvider.GetService<ICosmosDistributedLockStore>();
```

Comment:
Just an obverservation. Should the consumer need to know about ICosmosDistributedLockStore to instantiate a new lock? This should be an implementation detail abstracted from the client, in my opinion. I would rather see as usage pattern something like this:
lockFactory=_serviceProvider.GetService<IDistributedLockFactory>(); (factory internally depends on the store)
Func<Task> act = async () =>
            {
                using (await lockFactory.Create("test1"))
                {
                    // create a recurring job
                }
            };
Any toughts?

Reply:
Oh yes, that was the idea. I have fixed the tests now.